### PR TITLE
Do not export null as module.exports in amd.js. Resolves #2332.

### LINKED ISSUE #2332: SystemJS Type errors while loading the MFE (6.9.0 breaks named-export behavior)
@guybedford , @joeldenning ,

Our application is on Angular and we are using Single-spa.
We have recently updated SYSTEM JS version to 6.9.0 from 6.3.3 and deployed to Production. Since then we are seeing below new issue cropped up intermittently and blocking nearly 5% of users loading the application. 
This issue is from https://github.com/systemjs/systemjs/blob/master/src/extras/named-exports.js line number 33-40

TypeError: Cannot read property '__useDefault' of null Stack:TypeError: Cannot read property '__useDefault' of null
TypeError: null is not an object (evaluating 't.__useDefault') Stack:https://creta.state.com/Lib/systemJS.6.9.0.min.js

Can you please suggest me the possible resolution?

### PR DESCRIPTION
**Pull request by joeldenning**
See #2332 - both system-core and named-exports do not work if `_export(null)` occurs, as it appears to be invalid. However, amd.js would do this as explained in #2332. My fix here is specific only to null, rather than an object check + null check, since it's valid to loop over the keys of functions for re-exporting.

### CODE REVIEW

**Comment by joeldenning** [comment]
Done!

**Review by saak23 — APPROVED** [review]
Looks Good

File: src/extras/named-exports.js [hunk 1]
```diff
@@ -30,7 +30,7 @@
       // hook the _export function to note the default export
       var defaultExport, hasDefaultExport = false;
       var declaration = registerDeclare.call(this, function (name, value) {
-        if (typeof name === 'object' && name.__useDefault)
+        if (typeof name === 'object' && name && name.__useDefault)
```
Comment (guybedford) [inline src/extras/named-exports.js:33]: Good catch.

File: src/extras/amd.js [hunk 2]
```diff
@@ -56,7 +56,9 @@ import { errMsg } from '../err-msg.js';
           var amdResult = amdExec.apply(exports, depModules);
           if (amdResult !== undefined)
             module.exports = amdResult;
-          _export(module.exports);
+          // https://github.com/systemjs/systemjs/issues/2332
+          if (module.exports !== null)
+            _export(module.exports);
```
Comment (guybedford) [inline src/extras/amd.js:61]: With the core change we shouldn't need this null check anymore.

Comment (guybedford) [inline src/extras/amd.js:61]: With the core change we shouldn't need this null check anymore:

```suggestion
```

**Review by guybedford — APPROVED** [review]

**Pull request merged**